xicam/core/data/__init__.py

```diff
@@ -1,62 +1,53 @@
 from pathlib import Path
-from typing import Union, List
+from typing import Union, List, Set
 
 from qtpy.QtCore import Signal, QObject
 
 from databroker.utils import ALL
 from databroker.in_memory import BlueskyInMemoryCatalog
-from databroker.core import Header
 from warnings import warn
 
-
-# extension_map = {EDFPlugin: ['edf']}
-
-
 import mimetypes
 import warnings
 
 import entrypoints
 
 
-def ingest_file(filename):
-    """
-    Take in a file name; return a generator that yields (name, doc) pairs.
-    """
-    mimetype = detect_mimetype(filename)
-    ingestor = choose_ingestor(filename, mimetype)
-    generator_instance = ingestor(filename)
-    return generator_instance
-
-
-def detect_mimetype(filename):
+def detect_mimetypes(filename: str) -> List[str]:
     """
     Take in a filename; return a mimetype string like 'image/tiff'.
     """
     # First rely on custom "sniffers" that can employ file signatures (magic
     # numbers) or any other format-specific tricks to extract a mimetype.
+    matched_mimetypes = list()
+
     if Path(filename).is_dir():
-        return None  # TODO: do directories have mime-type?
+        return matched_mimetypes  # TODO: do directories have mime-type?
 
     with open(filename, 'rb') as file:
```

Comment:
We should probably handle when an invalid `filename` is passed

Reply:
Added higher-level sanity checks